tools/ci_build/github/azure-pipelines/win-wasm-browserstack-ci-pipeline.yml

```diff
@@ -0,0 +1,193 @@
+  - script: |
+     git submodule sync -- cmake/external/onnx
+     git submodule update --init -- cmake/external/onnx
+    workingDirectory: '$(Build.SourcesDirectory)'
+    displayName: 'Checkout submodule onnx'
+  - task: NodeTool@0
+    inputs:
+      versionSpec: '14.x'
+  - task: DownloadPipelineArtifact@2
+    inputs:
+      source: 'specific'
+      project: 'onnxruntime'
+      pipeline: 161
+      preferTriggeringPipeline: true
+      patterns: 'Release_*/**/*'
+      path: $(Pipeline.Workspace)/artifacts
+    displayName: 'Download WebAssembly artifacts'
+  - task: CopyFiles@2
+    inputs:
+      sourceFolder: $(Pipeline.Workspace)/artifacts
+      contents: |
+        **/*.wasm
+      targetFolder: $(Build.SourcesDirectory)/js/web/dist
+      flattenFolders: true
+    displayName: 'Binplace dist files'
+  - task: CopyFiles@2
+    inputs:
+      sourceFolder: $(Pipeline.Workspace)/artifacts
+      contents: |
+        **/*.js
+      targetFolder: $(Build.SourcesDirectory)/js/web/lib/wasm/binding
+      flattenFolders: true
+    displayName: 'Binplace js files'
+  - script: |
+     npm ci
+    workingDirectory: '$(Build.SourcesDirectory)/js'
+    displayName: 'npm ci /js/'
+  - script: |
+     npm ci
+    workingDirectory: '$(Build.SourcesDirectory)/js/common'
+    displayName: 'npm ci /js/common/'
+  - script: |
+     npm ci
+    workingDirectory: '$(Build.SourcesDirectory)/js/web'
+    displayName: 'npm ci /js/web/'
+  - script: |
+     npm run build
+    workingDirectory: '$(Build.SourcesDirectory)/js/web'
+    displayName: 'Build ort-web'
+  - task: BrowserStackConfig@0
+    inputs:
+      BrowserStackServiceEndPoint: 'BrowserStack Connection'
+      browserstackLocal: true
+    displayName: 'BrowserStack configuration setup'
+    timeoutInMinutes: 20
+  - script: |
+      export ONNXJS_TEST_BS_BROWSERS=BS_MAC_11_Safari_14,BS_MAC_11_Chrome_91,BS_ANDROID_11_Pixel_5
+      npm test -- suite0 --env=bs --wasm-init-timeout=30000 --file-cache
+    workingDirectory: '$(Build.SourcesDirectory)/js/web'
+    displayName: 'npm test (Suite0, BS_ANDROID, BS_MAC)'
+    env:
+      BROWSERSTACK_ACCESS_KEY: $(BROWSERSTACK_ACCESS_KEY)
+      BROWSERSTACK_USERNAME: $(BROWSERSTACK_USERNAME)
+  - script: |
+      export ONNXJS_TEST_BS_BROWSERS=BS_IOS_14_iPhoneXS
+      npm test -- suite1 --env=bs --wasm-init-timeout=30000 --file-cache --backend=wasm
+    workingDirectory: '$(Build.SourcesDirectory)/js/web'
+    displayName: 'npm test (Suite1, BS_IOS)'
+    env:
+      BROWSERSTACK_ACCESS_KEY: $(BROWSERSTACK_ACCESS_KEY)
+      BROWSERSTACK_USERNAME: $(BROWSERSTACK_USERNAME)
+  - task: BrowserStackStopLocal@0
+  - task: BrowserStackResults@0
+    displayName: 'BrowserStack results'
+    continueOnError: true
+    timeoutInMinutes: 10
+  - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
+    displayName: 'Clean Agent Directories'
+    condition: always()
+
+- job: build_onnxruntime_web_windows
```

Comment:
I think this job is not necessary for browserstack.

Reply:
this job is not running on BrowserStack, but on different agent pool, which supports Firefox and Edge.